LetsDoIt.CustomValueTypes/src/LetsDoIt.CustomValueTypes/Email.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Net.Mail;
+using System.Text.Json.Serialization;
+
+namespace LetsDoIt.CustomValueTypes
+{
+
+    [JsonConverter(typeof(EmailJsonConverter))]
+    public readonly struct Email
+    {
+        private readonly string _value;
+
+        public Email(string value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value), "Email can not be null!");
+            }
+
+            if (!IsValid(value))
+            {
+                throw new ArgumentException("Invalid email address.", value);
+            }
+
+            _value = value.ToLowerInvariant();
+        }
+
+        public static bool IsValid(string email)
+        {
+            if (string.IsNullOrEmpty(email))
+            {
+                return false;
+            }
+
+            try
+            {
+                var mailAddress = new MailAddress(email);
+
+                return string.Equals(mailAddress.Address, email, StringComparison.InvariantCultureIgnoreCase);
+            }
+            catch
+            {
+                return false;
+            }
+
+        }
+
+        public static bool TryParse(string candidate, out Email? email)
+        {
+            email = null;
+
+            if (string.IsNullOrWhiteSpace(candidate))
+            {
+                return false;
+            }
+
+            email = new Email(candidate);
+
+            return true;
+        }
+
+        public static Email Parse(string candidate)
+        {
+            if (string.IsNullOrWhiteSpace(candidate))
+            {
+                throw new ArgumentException("Email can not be empty!");
+            }
+
+            return new Email(candidate);
+        }
+
+        public static implicit operator string(Email email)
```

Comment:
This one should be explicit. 
The user should use it like below.
`Email mail = "mail@mail.com";`
`var mailAddress = mail.ToString();` or `var mailAddress = (string)mail;`

Reply:
Why not leave it as implicit. You can do it like this now.
```
 Email = email
```
Email is Email type
email is string type
when implicit you can directly assign it.
when explicit you need to put (string)email cast before it.

Reply:
it's up to you. I haven't seen it implicit before and I haven't seen any suggestion about making it implicit. Meanwhile, to be honest, the comments could not convince my mind using explicit for a particular reason.
https://stackoverflow.com/a/8041141/1695534